demo/src/main/java/com/novoda/landingstrip/SimpleTextTabActivity.java

```diff
@@ -39,8 +39,8 @@ public void onTabClicked(int position, Data data) {
             }
         };
         ExampleTabAdapter tabAdapter = new ExampleTabAdapter(getLayoutInflater(), setCurrentItemOnClick, data);
-
-        viewPager.addOnPageChangeListener(landingStrip);
+        ViewPager.OnPageChangeListener landingStripPageChanger = LandingStripPageChangeListener.create(landingStrip);
+        viewPager.addOnPageChangeListener(landingStripPageChanger);
```

Comment:
I like the first one , buy we don't want `LandingStrip` to become a `GodClass` so if it adds lots of complexity I'm happy with second approach that returns some other object